workshop/3-Experimentation/3.5-Amplitude-Performance-Metrics.ipynb

```diff
@@ -0,0 +1,461 @@
+    "\n",
+    "Retail Demo Store sends the following events to Personalize and Amplitude:\n",
+    "\n",
+    "|||||\n",
+    "|--- |--- |--- |--- |\n",
+    "|Event Name|Event Description|Personalize Training|Amplitude|\n",
+    "|ProductAdded|Every time a user adds a product to their cart; tracks the ID of the product that was added|X|X|\n",
+    "|ProductRemoved|Every time a user removes a product from their cart||X|\n",
+    "|ProductQuantityUpdated|Every time a user adds an additional unit of a product|X|X|\n",
+    "|ProductViewed|Every time a user views a product detail page|X|X|\n",
+    "|CartViewed|Every time a user views the contents of their shopping cart|X|X|\n",
+    "|CheckoutStarted|Every time a user begins the checkout process|X|X|\n",
+    "|OrderCompleted|Every time a user completes a purchase successfully|X|X|\n",
+    "\n",
+    "## Measuring Conversion Before Personalization is Deployed\n",
+    "\n",
+    "Because Amplitude reports rely on real user data, you will be using a data simulator to create enough events for you to be able to run a useful report in Amplitude.  The data simulator included in the Retail Demo Store will send the events shown above as though they were being generated by real users of the Retail Demo Store.  This will simulate common browsing behaviors at a sufficent volume to allow you to run a report that looks like one that might be used in your own production applications.\n",
+    "\n",
+    "The first simulated set of data will represent the Retail Demo Store prior to the deployment of the three personalization features you deployed in the first workshop exercise.\n",
+    "\n",
+    "Before you run the code, let's get your Amplitude key from SSM.  This will allow you to send simulated data directly to Amplitude.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "\n",
+    "ssm = boto3.client('ssm')\n",
+    "\n",
+    "response = ssm.get_parameter(\n",
+    "    Name='retaildemostore-amplitude-api-key'\n",
+    ")\n",
+    "\n",
+    "amplitude_config = {\n",
+    "    'api_key': response['Parameter']['Value']     # Do Not Change\n",
+    "}\n",
+    "\n",
+    "print(amplitude_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the cell above returned a valid Amplitude API key, you can run the following cell to populate your Ampliutude project with data.\n",
+    "\n",
+    "If you did not get a valid API key, it is likely you did not configure it in your deployment of Retail Demo Store.  Please go back and re-deploy Retail Demo Store with a valid Amplitude API key for the Amplitude project you plan to use for this workshop."
```

Comment:
Amplitude allows users to create different projects in an account - each project has its own API key and contains separate data repos, so each user in a workshop could deploy their own instance of Retail DS and have it point their own project - I mention this at the start of the workshop in the prerequisites section.  I do think that it would be useful to be able to re-deploy with a different API key after an instance of Retail DS is already deployed, though.  Maybe we tackle this in a separate PR?

Reply:
Also - moved the generate_users_json.py file to use the datagenerator library.  Also addressed your other comments.  This PR is getting quite complicated so probably should start splitting some of these changes out.

Reply:
> Amplitude allows users to create different projects in an account - each project has its own API key and contains separate data repos, so each user in a workshop could deploy their own instance of Retail DS and have it point their own project - I mention this at the start of the workshop in the prerequisites section. 

I was primarily thinking of the EE situation here where the accounts and repo are pre-deployed for each attendee.

> I do think that it would be useful to be able to re-deploy with a different API key after an instance of Retail DS is already deployed, though. Maybe we tackle this in a separate PR?

There is code that triggers a web-ui rebuild in personalize-pre-create-campaigns.py that could be copied into a module under the workshop directory and could be used by any workshop. Besides this workshop, the Personalization workshop could use it too. Created #44 to track this.